### PR TITLE
[MunicipalSpace] add candidate filters

### PR DESCRIPTION
**Pull request by ottaviano**

### CODE REVIEW

File: src/ApplicationRequest/Filter/ListFilter.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        return $this->tag;
+    }
+
+    public function setTag(?ApplicationRequestTag $tag): void
+    {
+        $this->tag = $tag;
+    }
+
+    public function getIsAdherent(): ?bool
```
Comment (Remg) [inline src/ApplicationRequest/Filter/ListFilter.php]: `function isAdherent()` ?

File: src/Form/ApplicationRequest/ApplicationRequestListFilterType.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefaults([
+                'data_class' => ListFilter::class,
+                'extended' => false,
+            ])
+            ->setAllowedTypes('extended', ['boolean'])
```
Comment (Remg) [inline src/Form/ApplicationRequest/ApplicationRequestListFilterType.php]: `->setAllowedTypes('extended', 'bool')` ?

File: src/Repository/ApplicationRequest/AbstractApplicationRequestRepository.php [hunk 3]
```diff
@@ -133,4 +140,64 @@ private function addFavoriteCitiesCondition(array $inseeCodes, QueryBuilder $qb)
...
+            $qb
+                ->leftJoin("${alias}.tags", 'tag_for_search')
+                ->andWhere('tag_for_search = :tag')
+                ->setParameter('tag', $filter->getTag())
+            ;
+        }
+
+        if ($filter->getTheme()) {
+            $qb
+                ->innerJoin("${alias}.favoriteThemes", 'theme')
```
Comment (Remg) [inline src/Repository/ApplicationRequest/AbstractApplicationRequestRepository.php]: `->innerJoin("$alias.favoriteThemes", 'theme')` ?

File: src/Form/ApplicationRequest/ApplicationRequestListFilterType.php (hunk 2 above)
Comment (ottaviano) [reply src/Form/ApplicationRequest/ApplicationRequestListFilterType.php]: even if it works well thanks to Type aliases I'll change it:
 
https://github.com/symfony/symfony/blob/bcc6d31ce439aeb625fba3756a79b7bfa7076c9c/src/Symfony/Component/OptionsResolver/OptionsResolver.php#L106-L110

File: src/ApplicationRequest/Filter/ListFilter.php (hunk 1 above)
Comment (ottaviano) [reply src/ApplicationRequest/Filter/ListFilter.php]: done

**Pull request merged**